OpenRA.Mods.Common/SpriteLoaders/PngSheetLoader.cs

```diff
@@ -102,12 +102,20 @@ void RegionsFromFrames(Png png, out List<Rectangle> regions, out List<float2> of
 			{
 				// Format: x,y,width,height;offsetX,offsetY
 				var coords = frame.Split(';');
-				regions.Add(FieldLoader.GetValue<Rectangle>("Region", coords[0]));
+				var region = FieldLoader.GetValue<Rectangle>("Region", coords[0]);
+				if (!new Rectangle(0, 0, png.Width, png.Height).Contains(region))
```

Comment:
The png rectangle doesn't change, so this definition can be pulled outside the loop.